README.md

```diff
@@ -2,880 +2,78 @@
-
-- *Default*: false
-
-service_ensure
---------------
-Ensure SSH service is running. Valid values are 'stopped' and 'running'.
-
-- *Default*: 'running'
-
-service_name
-------------
-Name of the SSH service.
-
-- *Default*: Based on OS
-
-service_enable
---------------
-Start SSH at boot. Valid values are 'true', 'false' and 'manual'.
-
-- *Default*: 'true'
-
-service_hasrestart
-------------------
-Specify that the init script has a restart command. Valid values are 'true' and 'false'.
-
-- *Default*: 'true'
-
-service_hasstatus
------------------
-Boolean to declare whether the service's init script has a functional status command.
-
-- *Default*: 'USE_DEFAULTS'
-
-ssh_key_ensure
---------------
-Export node SSH key. Valid values are 'present' and 'absent'.
-
-- *Default*: 'present'
-
-ssh_key_import
---------------
-Import all exported node SSH keys. Valid values are 'true' and 'false'.
-
-- *Default*: 'true'
-
-ssh_key_type
-------------
-Encryption type for SSH key. Valid values are 'ecdsa-sha2-nistp256', 'rsa', 'dsa', 'ssh-dss' and 'ssh-rsa'
-
-- *Default*: 'ssh-rsa'
-
-ssh_config_global_known_hosts_file
-----------------------------------
-File of the global known_hosts file
-
-- *Default*: '/etc/ssh/ssh_known_hosts'
-
-ssh_config_global_known_hosts_list
-----------------------------------
-Array of additional known_hosts files to be added to GlobalKnownHostsFile
-option together with `ssh_config_global_known_hosts_file`.
-
-- *Default*: undef
-
-ssh_config_global_known_hosts_owner
-----------------------------------
-Owner of the global known_hosts file
-
-- *Default*: 'root'
-
-ssh_config_global_known_hosts_group
-----------------------------------
-Group of the global known_hosts file
-
-- *Default*: 'root'
-
-ssh_config_global_known_hosts_mode
-----------------------------------
-File mode of the global known_hosts file
-
-- *Default*: '0644'
-
-ssh_config_user_known_hosts_file
---------------------------------
-Array of user's known_hosts files used in the ssh config option
-UserKnownHostsFile.
-
-- *Default*: undef
-
-manage_root_ssh_config
-----------------------
-Manage SSH config of root. Valid values are 'true' and 'false'.
-
-- *Default*: 'false'
-
-root_ssh_config_content
------------------------
-Content of root's ~/.ssh/config.
-
-- *Default*: "# This file is being maintained by Puppet.\n# DO NOT EDIT\n"
-
-manage_service
---------------
-Manage the sshd service through this module or not.  Valid values are 'true' and 'false'.
-
-- *Default*: 'true'
-
-===
 # Manage user's ssh_authorized_keys
-This works by passing the ssh::keys hash to the ssh_authorized_keys type with create_resources(). Because of this, you may specify any valid parameter for ssh_authorized_key. See the [Type Reference](http://docs.puppetlabs.com/references/stable/type.html#ssh_authorized_key) for a complete list.
+This works by passing the ssh::keys hash to the ssh_authorized_keys type with create_resources().
```

Comment:
We dont actually use the old `create_resources()` anymore thanks to you changing it to native iteration.

Reply:
👍 good point, also fixed the URL